plugins/dubbo/src/main/java/com/megaease/easeagent/plugin/dubbo/ApacheDubboCtxUtils.java

```diff
@@ -0,0 +1,197 @@
+		if (isConsumer) {
+			String groupName = requestUrl.getParameter(CommonConstants.GROUP_KEY);
+			String version = requestUrl.getParameter(CommonConstants.VERSION_KEY);
+
+			ApacheDubboClientRequest apacheDubboClientRequest = new ApacheDubboClientRequest(invoker, invocation);
+			RequestContext requestContext = context.clientRequest(apacheDubboClientRequest);
+			Span span = requestContext.span().start();
+			span.kind(apacheDubboClientRequest.kind());
+			span.name(apacheDubboClientRequest.name());
+			span.remoteServiceName(CommonConstants.DUBBO);
+			span.remoteIpAndPort(requestUrl.getIp(), requestUrl.getPort());
+
+			String argsList = config.argsCollectEnabled() ? JsonUtil.toJson(invocation.getArguments()) : null;
+			span.tag(DubboTags.CLIENT_APPLICATION.name, applicationName);
+			span.tag(DubboTags.GROUP.name, groupName);
+			span.tag(DubboTags.SERVICE.name, requestUrl.getPath());
+			span.tag(DubboTags.METHOD.name, method(invocation));
+			span.tag(DubboTags.SERVICE_VERSION.name, version);
+			span.tag(DubboTags.ARGS.name, argsList);
+
+			context.put(CLIENT_REQUEST_CONTEXT, requestContext);
+		} else {
+			ApacheDubboServerRequest apacheDubboServerRequest = new ApacheDubboServerRequest(invoker, invocation);
+			RequestContext requestContext = context.serverReceive(apacheDubboServerRequest);
+
+			Span span = requestContext.span().start();
+			span.kind(apacheDubboServerRequest.kind());
+			span.name(apacheDubboServerRequest.name());
+			span.remoteServiceName(CommonConstants.DUBBO);
+			span.remoteIpAndPort(rpcContext.getRemoteHost(), rpcContext.getRemotePort());
+			span.tag(DubboTags.SERVER_APPLICATION.name, applicationName);
+
+			context.put(SERVICE_REQUEST_CONTEXT, requestContext);
+		}
+	}
+
+	public static void finishSpan(URL url, DubboTraceConfig config, Context context, Result result, Throwable throwable) {
+		boolean isConsumer = url.getParameter(SIDE_KEY).equals(CONSUMER_SIDE);
+		if (result instanceof AsyncRpcResult) {
+			result.whenCompleteWithContext(new DubboAsyncCallback(isConsumer, config, context, throwable));
+		} else {
+			doFinishSpan(isConsumer, config, context, result, throwable);
+		}
+	}
+
+	public static void doFinishSpan(boolean isConsumer, DubboTraceConfig dubboTraceConfig, Context context, Result result, Throwable throwable) {
+		RequestContext requestContext;
+		if (isConsumer) {
+			requestContext = context.remove(CLIENT_REQUEST_CONTEXT);
+		} else {
+			requestContext = context.remove(SERVICE_REQUEST_CONTEXT);
+		}
+
+		if (requestContext == null) {
+			return;
+		}
+		try {
+			Span span = requestContext.span();
+			if (throwable != null) {
+				span.error(throwable);
+			}
+			if (result != null) {
+				if (result.getException() != null) {
+					span.error(result.getException());
+				} else {
+					span.tag(DubboTags.RESULT.name, dubboTraceConfig.argsCollectEnabled() ? JsonUtil.toJson(result.getValue()) : null);
+				}
+			}
```

Comment:
Please add the enabled configuration of the result tag, like `args.collect.enabled`, and the default is `false`.  
In many cases, the return value of Dubbo will be very large, which will affect the performance of the entire application

Reply:
I know there will be performance issues, so I configure `args.collect.enabled` to control the enablement of result tags and args tags to facilitate troubleshooting

Reply:
> I know there will be performance issues, so I configure `args.collect.enabled` to control the enablement of result tags and args tags to facilitate troubleshooting

Oh, the idea is very good, but args means parameters. It is better to define parameters and results separately, so you should split them into two configurations.